File: server/src/Luyenthi.Domain/Matrixs/Unit.cs
using Luyenthi.Domain.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Luyenthi.Domain
{
    public class Unit : IEntity<Guid>, IBaseEntity
    {
        public string Name { get; set; }
        public Guid ChapterId { get; set; }
        public Chapter Chapter { get; set; }
        public Guid Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public Guid? CreatedBy { get; set; }
        public Guid? UpdatedBy { get; set; }
        public virtual List<Question> Questions { get; set; }
        public virtual List<TemplateQuestion> TemplateQuestions { get; set; }
    }
}

File: client/src/models/matrix/Unit.ts
import { Chapter } from "./Chapter";
import { TemplateQuestion } from "./TemplateQuestion";

export interface Unit {
  name: string;
  code: string;
  id: string;
  templateQuestions?: TemplateQuestion[];
}
export interface UnitCreate {
  name: string;
  chapterId: string;
}
export interface UnitDetail {
  id: string;
  name: string;
  chapter: Chapter;
}
